### PR TITLE
docs: update HttpInterceptor API docs

### PR DESCRIPTION
**Pull request by sreevani-ship-it**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

API documention for interceptots not detailed enough
Issue Number: N/A

## What is the new behavior?
Updated Http Interceptors API doc

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jenniferfell** [comment]
Internal link: https://angular-team.atlassian.net/browse/DOCS-535

File: packages/common/http/src/interceptor.ts [hunk 1]
```diff
@@ -14,32 +14,28 @@ import {HttpRequest} from './request';
 import {HttpEvent} from './response';
 
 /**
- * Intercepts `HttpRequest` and handles them.
+ * Intercepts `HttpRequest` or `HttpRepsonse` and handles them.
  *
- * Most interceptors will transform the outgoing request before passing it to the
+ * Most interceptors transforms the outgoing request before passing it to the
  * next interceptor in the chain, by calling `next.handle(transformedReq)`.
+ * An interceptor may choose to transform the
```
Comment (CaerusKaru) [inline packages/common/http/src/interceptor.ts]: No anthropomorphizing. `An interceptor may transform the response event stream as well`. Here are throughout.

**Review by CaerusKaru — CHANGES_REQUESTED** [review]
Minor language change, otherwise LGTM

File: packages/common/http/src/interceptor.ts [hunk 2]
```diff
@@ -14,32 +14,28 @@ import {HttpRequest} from './request';
...
+ * response event stream as well, by applying additional Rx operators on the stream
```
Comment (trotyl) [inline packages/common/http/src/interceptor.ts]: `Rx` -> `RxJS` for consistency?

File: packages/common/http/src/interceptor.ts [hunk 3]
```diff
@@ -14,32 +14,28 @@ import {HttpRequest} from './request';
...
-   * Typically an interceptor will transform the outgoing request before returning
-   * `next.handle(transformedReq)`. An interceptor may choose to transform the
-   * response event stream as well, by applying additional Rx operators on the stream
-   * returned by `next.handle()`.
-   *
-   * More rarely, an interceptor may choose to completely handle the request itself,
-   * and compose a new event stream instead of invoking `next.handle()`. This is
-   * acceptable behavior, but keep in mind further interceptors will be skipped entirely.
+   * * **req**: The outgoing request to handle
+   * * **next**: The next interceptor in the chain
```
Comment (trotyl) [inline packages/common/http/src/interceptor.ts]: `or the real backend`?

**Review by benlesh — APPROVED** [review]

**Comment by benlesh** [comment]
Caretaker: The CircleCI test appears to be unrelated or a flake. No presubmit required, it's a docs only change.

File: packages/common/http/src/interceptor.ts [hunk 4]
```diff
@@ -14,32 +14,38 @@ import {HttpRequest} from './request';
 import {HttpEvent} from './response';
 
 /**
- * Intercepts `HttpRequest` and handles them.
+ * Intercepts `HttpRequest` or `HttpRepsonse` and handles them.
```
Comment (kara) [inline packages/common/http/src/interceptor.ts]:
```suggestion
 * Intercepts `HttpRequest` or `HttpResponse` and handles them.
```

File: packages/common/http/src/interceptor.ts [hunk 5]
```diff
@@ -14,32 +14,38 @@ import {HttpRequest} from './request';
...
+ *
+ * @usageNotes
+ *
+ * To use the same instance of `HttpInterceptors` for the entire app, import the `HttpClientModule`
+ * only in your `AppModule`, and add the interceptors to the root application injector .
+ * If you import `HttpClientModule` multiple times across different modules (for example, in lazy
+ * loading modules), each import creates a new copy of the `HttpClientModule`, which overwrites the interceptors
+ * provided in the root module.
  *
- * @publicApi
```
Comment (kara) [inline packages/common/http/src/interceptor.ts]: This `@publicAPI` notation needs to be here or it's removed from the public API (this is why it's failing the public API guard check on Circle CI). Can you re-add?

**Review by kara — CHANGES_REQUESTED** [review]
The public API error on Circle CI is a legitimate error (not a flake) because a `@publicAPI` marker was removed. This needs to be fixed before it can be merged.

**Comment by kara** [comment]
@chembu It looks like the CI error is fixed, but the typo still needs to be fixed.

**Review by kara — APPROVED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:229290380:BASE:229290402:1547516906861:36fda65d)

**Comment by AndrewKushnir** [comment]
@kara I've rebased this PR on top of the latest master and run g3 presubmit again to verify everything looks good. Could you please have a quick look at the changes to make sure there are no unexpected changes after rebase and return "merge" label back if everything is ok? Thank you.

**Pull request closed without merging**